airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -76,6 +76,38 @@ class PreemptibilityType(Enum):
     PREEMPTIBILITY_UNSPECIFIED = "PREEMPTIBILITY_UNSPECIFIED"
 
 
+class InstanceSelection:
```

Comment:
Could these classes be simplified as dataclasses? Airflow's minimum supported python is 3.8 so it should be safe to use them here.

Reply:
Done